drivers/dac/dac_stm32.c

```diff
@@ -120,6 +127,11 @@ static int dac_stm32_init(const struct device *dev)
 		return -EIO;
 	}
 
+	/* configure pinmux */
+	if (cfg->pinctrl_len != 0U) {
+		stm32_dt_pinctrl_configure(cfg->pinctrl, cfg->pinctrl_len);
+	}
+
```

Comment:
@erwango @gmarull @ABOSTM What do you think for F1 series ?

I know that this code is finde for all of the boards you ported as they are not F1 based.

Although no STM32F1 based board have DAC enabled but could be DAC enabled in the future (stm32f103x) for example.

STM32F1 has special REMAP functions, if you need to, you could inspire from USART and PWM drivers which handle the specific case of F1 series. 

Maybe support might be added as need arises in the future ?

Reply:
I looked at the PWM implementation and googled a bit to find out the differences between the F1 remap and the other MCUs, but it seemed that this would only affect the "Alternative Function" mapping, not if a pin is in analog mode. But I might be wrong here... would be good to get a confirmation from ST guys :)

Reply:
Nevermind, looks like DAC output cannot be remapped for F1 series (no such option in the LL), and the **dac_out1_pa4** or **dac_out2_pa5** are declared as NO_REMAP in F1 pinctrl DT. So actually, no remap is necessary here. 
```
                        /* DAC_OUT */

			dac_out1_pa4: dac_out1_pa4 {
				pinmux = <STM32F1_PINMUX('A', 4, ANALOG, NO_REMAP)>;
			};

			dac_out2_pa5: dac_out2_pa5 {
				pinmux = <STM32F1_PINMUX('A', 5, ANALOG, NO_REMAP)>;
			};
```